### PR TITLE
Navbar dynamic buttons and role-based authorities

### PR DESCRIPTION
**Pull request by Andrew-Rocket**
Added dynamic buttons to the navbar, now navbar buttons changed from profile&logout to login&registration if user is not authenticated. 

Each user's role name is also converted to his authority with prefix "ROLE_", therefore new perms annotations for each of the role added. Now, you can manage access on role basis, 

Also Auth pages is redirecting you if already logined in.

### CODE REVIEW

**Review by SerhiiTsoi — APPROVED** [review]

File: src/main/java/com/softserve/itacademy/controller/UserController.java [hunk 1]
```diff
@@ -33,8 +34,22 @@ public UserController(UserService userService) {
...
+            return new ResponseEntity<>(IsAuthenticatedResponse.builder().exists(false).build(), OK);
+        }
+
+        return new ResponseEntity<>(IsAuthenticatedResponse.builder().exists(true).userId(user.getId()).build(), OK);
+    }
+
     @GetMapping("/me")
     public ResponseEntity<UserFullTinyProjection> findCurrentUser(@AuthenticationPrincipal User user) {
+
+        if(user==null) {
```
Comment (DavidBabiy) [inline src/main/java/com/softserve/itacademy/controller/UserController.java:49]: spaces before and after == ?

File: src/main/java/com/softserve/itacademy/controller/view/ViewController.java [hunk 2]
```diff
@@ -1,34 +1,50 @@
...
     }
 
     @GetMapping(path = "/registration", produces = MediaType.TEXT_HTML_VALUE)
-    public String registrationView() {
-        return "registration.html";
+    public String registrationView(Principal principal) {
+        if (principal == null) {
+            return "registration.html";
+        } else {
+            return "redirect:user";
```
Comment (DavidBabiy) [inline src/main/java/com/softserve/itacademy/controller/view/ViewController.java:37]: Can we also check if it is admin registered or logged in, and redirect him to admin panel?

File: src/main/java/com/softserve/itacademy/controller/UserController.java (hunk 1 above)
Comment (Andrew-Rocket) [reply src/main/java/com/softserve/itacademy/controller/UserController.java:49]: Sad, but not worth much attention. Fixed on local.

File: src/main/java/com/softserve/itacademy/controller/view/ViewController.java (hunk 2 above)
Comment (Andrew-Rocket) [reply src/main/java/com/softserve/itacademy/controller/view/ViewController.java:37]: We could, but I still think that we should redirect everybody to the last page user was at, or maybe to home page. Redirecting to panels sounds weird.

**Pull request merged**